### PR TITLE
Skip mock-recipe-server deploys when not configured

### PR DESCRIPTION
**Pull request by mythmon**

### CODE REVIEW

File: mock-recipe-server/bin/ci/deploy [hunk 1]
```diff
@@ -1,4 +1,11 @@
 #!/usr/bin/env bash
-set -eu
+set -e
+
+if [[ -z $MOCK_SERVER_ARTIFACTS || -z $MOCK_SERVER_S3_BUCKET ]]; then
```
Comment (Osmose) [inline mock-recipe-server/bin/ci/deploy]: Might wanna check for AWS credentials too.

Comment (mythmon) [inline mock-recipe-server/bin/ci/deploy]: Do you know what envvars AWS creds go in? Or can I ask the `aws` tool about that?

Comment (Osmose) [inline mock-recipe-server/bin/ci/deploy]: Ah, yeah. [`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-environment).

**Comment by mythmon** [comment]
I added more variables. I also tested this locally, and the envvar checking works (I didn't test upload).

**Pull request merged**